reference-adapter/adapter-master-data-service/src/main/java/org/egov/service/ProjectEnrichmentService.java

```diff
@@ -30,16 +33,58 @@ public void enrichProjectData(ProjectRequest projectRequest) {
 
         AuditDetails auditDetails = null;
 
+        String userId = (requestHeader.getUserInfo() != null ? requestHeader.getUserInfo().getUuid() : null);
         if (project.getAuditDetails() == null) {
             auditDetails = enrichAuditDetails
-                    .enrichAuditDetails(requestHeader.getUserInfo().getUuid(), project.getAuditDetails(), true);
+                    .enrichAuditDetails(userId, project.getAuditDetails(), true);
         } else {
             auditDetails = enrichAuditDetails
-                    .enrichAuditDetails(requestHeader.getUserInfo().getUuid(), project.getAuditDetails(), false);
+                    .enrichAuditDetails(userId, project.getAuditDetails(), false);
         }
 
         project.setId(UUID.randomUUID().toString());
         project.setAuditDetails(auditDetails);
     }
 
+    public Project enrichUpdateProjectData(ProjectRequest projectRequest) {
```

Comment:
Can you reuse the enrich create code? Create a new method that contains the common code.